### PR TITLE
B162-removed listeners for queue when leaving screen. Also updated join queue flow

### PR DESCRIPTION
**Pull request by JohnnyPersichilli**
Previously if the queue would delete if the user was previously the host of a queue but i forgot to check whether or not he was rejoining the same queue before deleting it.  The fix to this bug is also in this pr. Should have probably made another ticket for this

### CODE REVIEW

File: CranberryQueue/Map/MapController.swift [hunk 1]
```diff
@@ -68,27 +72,28 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
...
         
-        if( (self.queueId) != nil && !isHost ){
-            self.db?.collection("contributor").document(self.queueId!).collection("members").document(self.uid).delete()
-        }else if( (self.queueId) != nil && isHost){
-            self.db?.collection("contributor").document(self.queueId!).delete()
-            
-            self.db?.collection("song").whereField("queueId", isEqualTo: self.queueId).getDocuments(completion: { (snapshot, err) in
-                guard let snap = snapshot else {
-                    return
-                }
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: I moved the join queue stuff into the parent mapViewController so the vc presentation code would be in the same spot for host and guest. We have concurrent conflicts now. Next time you should let the group know before leaving the scope of your ticket.

File: CranberryQueue/Map/MapController.swift [hunk 2]
```diff
@@ -48,6 +48,10 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
         setupLocation()
     }
     
+    override func viewWillDisappear(_ animated: Bool) {
+        queuesInLocationRef?.remove()
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: Does this ever get set up again in viewWillAppear? Or will it not be listening to new queues when you return to the map once?

Comment (JohnnyPersichilli) [reply CranberryQueue/Map/MapController.swift]: yes i believe it gets called through getGeoCode which gets called from the location manager.  I confirmed this theory but putting a breakpoint on the ref in the function and it hits the breakpoint when returning from the queue to the map so i think it is correct.  We can confirm this in person though

File: CranberryQueue/Map/MapController.swift (hunk 1 above)
Comment (JohnnyPersichilli) [reply CranberryQueue/Map/MapController.swift]: do you want me to merge master into my branch and change it to update this bug fix and recommit?

**Pull request merged**